pkg/utils/helpers.go

```diff
@@ -4,15 +4,30 @@ import (
 	"crypto/sha256"
 	"encoding/hex"
 	"fmt"
+	"regexp"
 	"strings"
 
 	fluxhelmv2beta1 "github.com/fluxcd/helm-controller/api/v2beta1"
 	"helm.sh/helm/v3/pkg/chart"
 	"sigs.k8s.io/yaml"
 )
 
+const (
+	DNS1123NameMaximumLength    = 63
+	DNS1123NotAllowedChars      = "[^-a-z0-9]"
+	DNS1123NotAllowedStartChars = "^[^a-z0-9]+"
+)
+
 func ConvertToDNS1123(in string) string {
-	return strings.ReplaceAll(in, "_", "-")
+	in = strings.ToLower(in)
+
+	notAllowedChars := regexp.MustCompile(DNS1123NotAllowedChars)
+	in = notAllowedChars.ReplaceAllString(in, "-")
+
+	notAllowedStartChars := regexp.MustCompile(DNS1123NotAllowedStartChars)
+	in = notAllowedStartChars.ReplaceAllString(in, "a")
```

Comment:
can we replace it with an empty string `""`?

Reply:
Is empty string a valid name? For example, if name="?%$@!&" (all invalid chars) then first all chars will be replaced with "-" and then all "-" chars are replaced with "" because "-" is an invalid start char.

So, if we want to change "a" to "" then we can add some valid char in case of empty string.

Reply:
I see your point.
Another corner case that comes up is If we have two subcharts with names `%@$!` & `##@#` and `a` how would we differentiate between each of these? Might need a cleaner way with some unique identifier than just replacing with a char. A UUID based on the input name?

Reply:
Maybe the truncated hash could be an option?

Reply:
If after replace output string is empty then take hash of input string and truncate to 63 chars?

Reply:
> A UUID based on the input name?

Hash of the original input name will be great.

Reply:
> > A UUID based on the input name?
> 
> Hash of the original input name will be great.

Let's try that !